code/API_definitions/device-reachability-status.yaml

```diff
@@ -0,0 +1,397 @@
+
+    # API Functionality
+
+    The API exposes following capabilities:
+
+    ## Device Reachability situation
+
+    The endpoint `POST /retrieve-reachability-status` allows to get current connectivity status information synchronously.
+
+
+    ## Further info and support
+
+    (FAQs will be added in a later version of the documentation)
+
+  termsOfService: http://swagger.io/terms/
+  contact:
+    email: project-email@sample.com
+  license:
+    name: Apache 2.0
+    url: https://www.apache.org/licenses/LICENSE-2.0.html
+  version: wip
+externalDocs:
+  description: Product documentation at CAMARA
+  url: https://github.com/camaraproject/
+
+servers:
+  - url: "{apiRoot}/reachability-retrieval/v0"
+    variables:
+      apiRoot:
+        default: http://localhost:9091
+        description: API root
+tags:
+  - name: Device reachability status
+    description: Operations to get the current reachability status of a device
+paths:
+  /retrieve-reachability-status:
+    post:
+      tags:
+        - Device reachability status
+      summary: "Get the current reachability status information"
+      description: Get the current reachability status information
+      operationId: getReachabilityStatus
+      parameters:
+        - $ref: '#/components/parameters/x-correlator'
+      security:
+        - openId:
+            - reachability-retrieval:read
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: "#/components/schemas/RequestReachabilityStatus"
+        required: true
+      responses:
+        "200":
+          description: Contains information about current reachability status
+          headers:
+            x-correlator:
+              $ref: '#/components/headers/x-correlator'
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/ReachabilityStatusResponse"
+              examples:
+                Connected-With-SMS:
+                  value:
+                    lastStatusTime: "2024-02-20T10:41:38.657Z"
+                    reachabilityStatus: CONNECTED_SMS
+                Connected-With-DATA:
+                  value:
+                    lastStatusTime: "2024-02-20T10:41:38.657Z"
+                    reachabilityStatus: CONNECTED_DATA
+                Not-Connected:
+                  value:
+                    lastStatusTime: "2024-02-20T10:41:38.657Z"
+                    reachabilityStatus: NOT_CONNECTED
+        "400":
+          $ref: "#/components/responses/Generic400"
+        "401":
+          $ref: "#/components/responses/Generic401"
+        "403":
+          $ref: "#/components/responses/ReachabilityPermissionDenied403"
+        "404":
+          $ref: "#/components/responses/ReachabilityNotFound404"

```

Comment:
I can see that we have the generic NOT_FOUND in addition to the DeviceIdentifierNotFound. 
In which scenario does the API return an standard NOT_FOUND?

Reply:
I agree with @fernandopradocabrillo 

Maybe we should more go the way of having a reachability Status `UNKNOWN` with HTTP - 200 than a HTTP-404 for Reachability not found.

see https://github.com/camaraproject/DeviceStatus/issues/148

@bigludo7 wdyt?

Reply:
@maxl2287 @fernandopradocabrillo I can remove the NOT_FOUND. No problem.

BTW we're still waiting outcomes for #148 which is a fair point. I will comment there.